### DISCUSSION TITLE
How Do I Do Networking

### DISCUSSION THREAD

**Discussion by ghost** [CosmosOS/Cosmos#3117] [General]
I Need To Know How Do I Make A Browser, Ping Command Using ICMP, DNS Settings And FTP Support For Browser

**Comment by Gabolate** [comment]
Please check the [docs](https://github.com/CosmosOS/Cosmos/blob/master/Docs/articles/Kernel/Network.md) for more info about it

**Reply by ghost** [reply]
> Please check the [docs](https://github.com/CosmosOS/Cosmos/blob/master/Docs/articles/Kernel/Network.md) for more info about it

So I Did This And Got Ping failed: Arg_IndexOutOfRangeException

// from my kernel class
var nic = NetworkDevice.GetDeviceByName("eth0");
System.Console.WriteLine("Loaded Network Part 1 of 2.");
IPConfig.Enable(nic, new Address(192, 168, 1, 69), new Address(255, 255, 255, 0), new Address(192, 168, 1, 1));
System.Console.WriteLine("Loaded Network Part 2 of 2");

// form my command class
using (var icmpClient = new ICMPClient())
{
    var targetAddress = new Address(8, 8, 8, 8); // Google's public DNS as an example
    icmpClient.Connect(targetAddress);

    try
    {
        Console.WriteLine("Sending ICMP Echo Request...");
        icmpClient.SendEcho();

        var endpoint = new EndPoint(Address.Zero, 0);
        int elapsedTime = icmpClient.Receive(ref endpoint);

        Console.WriteLine($"ICMP Echo Reply received in {elapsedTime}ms from {endpoint.Address}");
    }
    catch (Exception ex)
    {
        Console.WriteLine($"Ping failed: {ex.Message}");
    }
}
return "";

**Reply by valentinbreiz** [reply]
> > Please check the [docs](https://github.com/CosmosOS/Cosmos/blob/master/Docs/articles/Kernel/Network.md) for more info about it
> 
> So I Did This And Got Ping failed: Arg_IndexOutOfRangeException
> 
> // from my kernel class
> var nic = NetworkDevice.GetDeviceByName("eth0");
> System.Console.WriteLine("Loaded Network Part 1 of 2.");
> IPConfig.Enable(nic, new Address(192, 168, 1, 69), new Address(255, 255, 255, 0), new Address(192, 168, 1, 1));
> System.Console.WriteLine("Loaded Network Part 2 of 2");
> 
> // form my command class
> using (var icmpClient = new ICMPClient())
> {
>     var targetAddress = new Address(8, 8, 8, 8); // Google's public DNS as an example
>     icmpClient.Connect(targetAddress);
> 
>     try
>     {
>         Console.WriteLine("Sending ICMP Echo Request...");
>         icmpClient.SendEcho();
> 
>         var endpoint = new EndPoint(Address.Zero, 0);
>         int elapsedTime = icmpClient.Receive(ref endpoint);
> 
>         Console.WriteLine($"ICMP Echo Reply received in {elapsedTime}ms from {endpoint.Address}");
>     }
>     catch (Exception ex)
>     {
>         Console.WriteLine($"Ping failed: {ex.Message}");
>     }
> }
> return "";

Can you debug where is the exception happening? Are using VMWare in NAT mode or bridge?

**Comment by ghost** [comment]


**Reply by valentinbreiz** [reply]
You have google :) Please don't expect others doing what you can do.

**Comment by ghost** [comment]
Uh I Think I Can Debug
One Sec

**Reply by valentinbreiz** [reply]
Could you please post your next message on the issue you created?
